app/controllers/timeline_entries_controller.rb

```diff
@@ -33,12 +33,14 @@ def update
 
   def destroy
     @timeline_entry = coronavirus_page.timeline_entries.find(params[:id])
+    backup = @timeline_entry.attributes.except("id", "created_at", "updated_at")
 
-    message = if @timeline_entry.destroy && draft_updater.send
-                { notice: "Timeline entry was successfully deleted." }
-              else
-                { alert: "Timeline entry couldn't be deleted" }
-              end
+    if @timeline_entry.destroy && draft_updater.send
```

Comment:
It seems like this destroy would reset the ordering of everything and the recreate would then move this item to the end?

I imagine the right thing to do is to put things into a transaction and abort the transaction if updating the Publishing API failed so the data remains intact.

Reply:
Good shout. I'll wrap the destroy and publishing-api calls in a transaction, and do the same for announcements and sub-sections too.